vmicore/src/include/vmicore/plugins/PluginInterface.h

```diff
@@ -23,35 +21,100 @@ namespace VmiCore::Plugin
 
+        /**
+         * Create a software breakpoint at the given virtual address. The breakpoint will be protected, so that
+         * it won't be visible to the guest through reading the memory. Multiple breakpoints per address are allowed and
+         * will not interfere with each other. If the breakpoint is hit, the supplied callback will be called.
+         *
+         * @param targetVA Target address to place the breakpoint on.
+         * @param processInformation The process information for the target process. Can be obtained via
+         * getRunningProcesses().
+         * @param callbackFunction It is recommended to create the lambda with the help of VMICORE_SETUP_MEMBER_CALLBACK
+         * from <a href="file:../callback.h">callback.h</a>
+         * @return Shared pointer to a breakpoint object. Can be used to delete the breakpoint.
+         */
         [[nodiscard]] virtual std::shared_ptr<IBreakpoint>
         createBreakpoint(uint64_t targetVA,
                          const ActiveProcessInformation& processInformation,
                          const std::function<BpResponse(IInterruptEvent&)>& callbackFunction) = 0;
 
+        /**
+         * Retrieves the path to the directory where plugins are supposed to store any files that are generated
+         * throughout the course of a run. However, it is generally discouraged to store files directly. Instead,
+         * the writeToFile or the logging APIs should be used.
+         */
         [[nodiscard]] virtual std::unique_ptr<std::string> getResultsDir() const = 0;
 
+        /**
+         * Creates a new logger with a given name.
+         */
         [[nodiscard]] virtual std::unique_ptr<ILogger> newNamedLogger(std::string_view name) const = 0;
 
+        /**
+         * Saves content to a file with the given name. Does not append. Saving the same file more than once is
+         * undefined behavior. Use the other overload for raw data.
+         */
         virtual void writeToFile(const std::string& filename, const std::string& message) const = 0;
 
+        /**
+         * Saves content to a file with the given name. Does not append. Saving the same file more than once is
+         * undefined behavior. Use the other overload for strings.
+         */
         virtual void writeToFile(const std::string& filename, const std::vector<uint8_t>& data) const = 0;
 
+        /**
+         * Only useful if using a gRPC connection, does nothing otherwise. Will send an error event via a separate
+         * channel which indicates that the run is not successful.
+         */
         virtual void sendErrorEvent(std::string_view message) const = 0;
 
+        /**
+         * Only useful if using a gRPC connection, does nothing otherwise. Will send an inmemory scanner detection event
+         * via a gRPC channel.
+         */
         virtual void sendInMemDetectionEvent(std::string_view message) const = 0;
```

Comment:
nit: Not part of this PR, but I think we should replace this with a more plugin agnostic version.

Reply:
Sure, you could create an issue for that.